musica/micm/micm.F90

```diff
@@ -1,48 +1,91 @@
-module micm
+module musica_ccpp_micm
   use iso_c_binding
 
   ! Note: "micm_core" is included in an external pre-built MICM library that the host
   ! model is responsible for linking to during compilation
-  use micm_core, only: micm_t
+  use musica_micm, only: micm_t
+  use musica_ccpp_util, only: has_error_occurred
   use ccpp_kinds, only: kind_phys
+  use musica_namelist, only: filename_of_micm_configuration
```

Comment:
So I realized this morning that you might have done it this way because you can't pass in extra variables into the register call (is that true @peverwhee)?  If so then I think keeping this `use` statement here is totally fine for this PR, but we on the CCPP-side will need to figure out a way to pass in configuration files into register phase subroutines like this one soon(-ish).

Reply:
@nusbaume - correct. we don't currently have metadata for the dynamic constituent routines.

Reply:
@peverwhee Thanks for the confirmation!  In that case I'll go ahead and mark this concern resolved (as I don't think we'll want to change it for this PR).  However, I made a note for our Wednesday meeting to discuss it, as it is definitely a capability I think we'll need long-term.